### PR TITLE
Feature/date picker

### PR DESCRIPTION
**Pull request by alkali47**
This is missing the technical section as well as the reference to it in the documentation because that is unavailable at the moment

### CODE REVIEW

File: articles/ds/components/date-picker/index.asciidoc [hunk 1]
```diff
@@ -41,3 +54,187 @@ The usage examples for this component are currently in progress. In the meantime
...
+[source, java]
+----
+include::{root}/src/main/java/com/vaadin/demo/component/datepicker/DatePickerIndividualInputFields.java[render,tags=snippet,indent=0,group=Java]
+----
+--
+
+=== Indicate the Date Format
+
+Use a placeholder or helper to indicate how the input should be formatted.
+For example, 12/6/2020 represents different dates for Americans and Europeans.
```
Comment (DiegoCardoso) [inline articles/ds/components/date-picker/index.asciidoc:206]: IMO, from all Vale errors, this is the only that should be ignored, while the other ones could be fixed.

**Comment by jouni** [comment]
Build error:

```
10:29:27 PM: ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-custom-validation.ts
10:29:27 PM: [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-custom-validation.ts(31,7)
10:29:27 PM:       TS6196: 'DatePickerWorkingDays' is declared but never used.
10:29:27 PM: Child HtmlWebpackCompiler:
10:29:27 PM:      1 asset
10:29:27 PM:     Entrypoint HtmlWebpackPlugin_0 = __child-HtmlWebpackPlugin_0
10:29:27 PM:     [0] ../node_modules/.pnpm/html-webpack-plugin@4.5.1_webpack@4.46.0/node_modules/html-webpack-plugin/lib/loader.js!./index.html 898 bytes {0} [built]
```

**Comment by jouni** [comment]
The build is failing because of a broken cross reference:

    12:02:17 AM: error Broken cross reference ../Overview in tools/mpr/configuration/session.asciidoc

**Comment by DiegoCardoso** [comment]
> Should be already fixed in the latest branch, so updating/rebasing should be enough.

Now it's updated with `latest`.

**Comment by jouni** [comment]
A bunch of build errors:

```
3:49:44 PM: [ERROR] Failed to execute goal com.vaadin:vaadin-maven-plugin:20.0.0.alpha3:build-frontend (default-cli) on project docs: Execution default-cli of goal com.vaadin:vaadin-maven-plugin:20.0.0.alpha3:build-frontend failed: Webpack process exited with non-zero exit code.
3:49:44 PM: [ERROR] Stderr: 'Hash: 699fea439839bb785bcf
3:49:44 PM: [ERROR] Version: webpack 4.46.0
3:49:44 PM: [ERROR] Time: 53746ms
3:49:44 PM: [ERROR] Built at: 03/16/2021 1:49:44 PM
3:49:44 PM: [ERROR]  9 assets
3:49:44 PM: [ERROR] Entrypoint bundle = VAADIN/build/vaadin-bundle-ac310f7530cf8cc74a1a.cache.js
3:49:44 PM: [ERROR] Entrypoint export = VAADIN/build/vaadin-export-8bcc4f2cbfc711bf9b97.cache.js
3:49:44 PM: [ERROR]   [0] ./generated/theme.js + 5 modules 4.67 KiB {0} {1} [built]
3:49:44 PM: [ERROR]       | ./generated/theme.js 112 bytes [built]
3:49:44 PM: [ERROR]       | ./themes/docs/docs.generated.js 2.97 KiB [built]
3:49:44 PM: [ERROR]       | ./themes/docs/styles.css 159 bytes [built]
3:49:44 PM: [ERROR]       | ./themes/docs/document.css 849 bytes [built]
3:49:44 PM: [ERROR]       |     + 2 hidden modules
3:49:44 PM: [ERROR]  [74] ../target/frontend/accordion-summary-wc.js 4.89 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [75] ../target/frontend/date-picker-date-format-indicator-wc.js 5.06 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [76] ../target/frontend/date-time-picker-week-numbers-wc.js 5.01 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [77] ../target/frontend/select-disabled-wc.js 4.86 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [78] ../target/frontend/select-basic-wc.js 4.82 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [79] ../target/frontend/date-time-picker-basic-wc.js 4.93 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [80] ../target/frontend/number-field-big-decimal-wc.js 4.96 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [81] ../target/frontend/number-field-stepper-controls-wc.js 5.02 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [82] ../target/frontend/accordion-content-wc.js 4.89 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [83] ../target/frontend/login-additional-information-wc.js 5.02 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [84] ../target/frontend/accordion-reverse-panels-wc.js 4.97 KiB {1} {2} [built]
3:49:44 PM: [ERROR]  [85] ../target/frontend/checkbox-adjacent-groups-wc.js 4.97 KiB {1} {2} [built]
3:49:44 PM: [ERROR] [246] ../target/frontend/generated-flow-imports.js-filtered.js 6.37 KiB {1} [built]
3:49:44 PM: [ERROR] [247] ./generated/vaadin.ts + 8 modules 135 KiB {0} [built]
3:49:44 PM: [ERROR]       | ./generated/vaadin.ts 79 bytes [built]
3:49:44 PM: [ERROR]       | ./index.ts 543 bytes [built]
3:49:44 PM: [ERROR]       | ../target/flow-frontend/index.ts 124 bytes [built]
3:49:44 PM: [ERROR]       | ./main-layout.ts 2.22 KiB [built]
3:49:44 PM: [ERROR]       | ../target/flow-frontend/ConnectionIndicator.ts 14.3 KiB [built]
3:49:44 PM: [ERROR]       | ../target/flow-frontend/Flow.ts 13.8 KiB [built]
3:49:44 PM: [ERROR]       | ../target/flow-frontend/Connect.ts 8.84 KiB [built]
3:49:44 PM: [ERROR]       | ../target/flow-frontend/Authentication.ts 4.03 KiB [built]
3:49:44 PM: [ERROR]       |     + 1 hidden module
3:49:44 PM: [ERROR]     + 430 hidden modules
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-auto-open.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-auto-open.ts(13,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
[ERROR]   Type 'null' is not assignable to type 'Node'.
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-custom-validation.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-custom-validation.ts(15,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
[ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-date-format-indicator.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-date-format-indicator.ts(13,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
[ERROR]   Type 'null' is not assignable to type 'Node'.
[ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-date-range.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-date-range.ts(14,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-individual-input-fields.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-individual-input-fields.ts(13,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-initial-position.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-initial-position.ts(14,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-min-max.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-min-max.ts(14,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
3:49:44 PM: [ERROR]
3:49:44 PM: [ERROR] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-week-numbers.ts
3:49:44 PM: [ERROR] [tsl] ERROR in /opt/build/repo/frontend/demo/component/datepicker/date-picker-week-numbers.ts(14,16)
3:49:44 PM: [ERROR]       TS2345: Argument of type 'ShadowRoot | null' is not assignable to parameter of type 'Node'.
3:49:44 PM: [ERROR]   Type 'null' is not assignable to type 'Node'.
```

**Comment by jouni** [comment]
A lot of lint errors now. Likely because of https://github.com/vaadin/docs/commit/862d9fe68ceb405430282b31f795bfc6626dc7ef

Make sure to merge/rebase from latest, and run `npm run lint`

**Comment by DiegoCardoso** [comment]
I don't understand why this commit (b75a257e6d25187e54496ec8d70bdf8defbb798f) merging `latest` onto this PR's branch is reverting the changes introduced by the prettier upgrade commit 🤔

**Comment by DiegoCardoso** [comment]
I had to rebase again and now it looks ok.

File: articles/ds/components/date-picker/index.asciidoc (hunk 1 above)
Comment (jouni) [reply articles/ds/components/date-picker/index.asciidoc:205]: Fixed

**Review by jouni — APPROVED** [review]

**Pull request merged**